packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -460,76 +460,114 @@ class IrisGridTableModel extends IrisGridModel {
-
-      let frontColumns = [];
-      let backColumns = [];
-      let frozenColumns = [];
-
-      if (hints?.frontColumns) {
-        frontColumns = hints.frontColumns
-          .map(name => columnMap.get(name))
-          .filter(Boolean);
-      }
-      if (hints?.backColumns) {
-        backColumns = hints.backColumns
-          .map(name => columnMap.get(name))
-          .filter(Boolean);
-      }
-      if (userFrozenColumns) {
-        frozenColumns = userFrozenColumns
-          .map(name => columnMap.get(name))
-          .filter(Boolean);
-      }
+  getColumnsWithLayoutHints = memoize(
+    (columnMap, frontColumns, backColumns, frozenColumns) => {
+      if (frontColumns.length || backColumns.length || frozenColumns.length) {
+        let finalFrontColumns = [];
+        let finalBackColumns = [];
+        let finalFrozenColumns = [];
+
+        if (frontColumns.length) {
+          finalFrontColumns = frontColumns
+            .map(name => columnMap.get(name))
+            .filter(Boolean);
+        }
+        if (backColumns.length) {
+          finalBackColumns = backColumns
+            .map(name => columnMap.get(name))
+            .filter(Boolean);
+        }
+        if (frozenColumns.length) {
+          finalFrozenColumns = frozenColumns
+            .map(name => columnMap.get(name))
+            .filter(Boolean);
+        }
 
-      if (
-        frontColumns.length !== (hints?.frontColumns?.length ?? 0) ||
-        backColumns.length !== (hints?.backColumns?.length ?? 0)
-      ) {
-        throw new Error(
-          'Layout hints are invalid (contain invalid column names)'
-        );
-      }
+        if (
+          finalFrontColumns.length !== frontColumns.length ||
+          finalBackColumns.length !== backColumns.length
+        ) {
+          throw new Error(
+            'Layout hints are invalid (contain invalid column names)'
+          );
+        }
 
-      const frontColumnSet = new Set(frontColumns);
-      const backColumnSet = new Set(backColumns);
-      const frozenColumnSet = new Set(frozenColumns);
-      const middleColumns = columns.filter(
-        col =>
-          !frontColumnSet.has(col) &&
-          !backColumnSet.has(col) &&
-          !frozenColumnSet.has(col)
-      );
+        const frontColumnSet = new Set(finalFrontColumns);
+        const backColumnSet = new Set(finalBackColumns);
+        const frozenColumnSet = new Set(finalFrozenColumns);
+        const middleColumns = [...columnMap.values()].filter(
+          col =>
+            !frontColumnSet.has(col) &&
+            !backColumnSet.has(col) &&
+            !frozenColumnSet.has(col)
+        );
 
-      return [
-        ...frozenColumns,
-        ...frontColumns,
-        ...middleColumns,
-        ...backColumns,
-      ];
+        return [
+          ...finalFrozenColumns,
+          ...finalFrontColumns,
+          ...middleColumns,
+          ...finalBackColumns,
+        ];
+      }
+      return this.table.columns;
```

Comment:
Should probably return `[...columnMap.values()]` to be safe. Could just initialize `const columns = [...columnMap.values();]` at the top and use here and when calculating `middleColumns`